File: src/common/models/dto/VacationRequestDto.ts
import { VacationRequestStatus } from "src/common/enums/VacationRequestStatus";

export interface VacationRequestDto {
  id: number;
  userId: number;
  userFullName: string;
  startDate: Date;
  endDate: Date;
  daysCount: number;
  statusId: VacationRequestStatus;
}

File: TwojUrlop.DomainModel/Common/Models/VacationRequestDto.cs
namespace TwojUrlop.DomainModel.Common.Models;
public class VacationRequestDto
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public virtual UserDto User { get; set; } = default!;
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public int DaysCount { get; set; }
    public int StatusId { get; set; }
}